src/base_circuitpython/neopixel_write.py

```diff
@@ -0,0 +1,54 @@
+# overriden neopixel_write library to write to frontend
+
+# original implementation docs for neopixel_write:
+# https://circuitpython.readthedocs.io/en/5.0.x/shared-bindings/neopixel_write/__init__.html
+
+
+import pathlib
+import sys
+import os
+
+from common import utils
+from adafruit_circuitplayground import cp
+import base_cp_constants as CONSTANTS
+
+
+def neopixel_write(gpio, buf):
+    """Write buf out on the given DigitalInOut."""
+
+    if len(tuple(buf)) > 0:
+        # if we are explicitly given
+        # the clue pin, that means that
+        # the clue is definitely the active device
+        # because the constructor for the
+        # clue is what calls neopixel
+        # with the clue pin argument
+        if gpio.pin != CONSTANTS.CLUE_PIN:

```

Comment:
Ah, I wasn't very clear with how I implemented this!

So in real life, both the CLUE and CPX use the neopixel library; however, when they were developing the CPX, they created pixel.py, which is pretty much an edited version of the neopixel library. 

In my implementation, my clue library uses the neopixel library, which means that all streams of neopixel data go through my neopixel_write function. They can also directly import the neopixel library and interact with it directly. These are both either directly or indirectly sending data through neopixel.py

At the same time, on the CPX, simulator users can now also directly import neopixel (new with the neopixel library) or uses the field in the cpx library (how we've always done it, and controlled in pixel.py). Therefore, pixel info from the CPX can come from neopixel.py or pixel.py.

Therefore, when using the neopixel library, I have it sending two messages: one to the CPX frontend and another to the CLUE backend. This can be a little inefficient, especially if the neopixel was called by the CLUE library in the first place! Therefore, I have my custom clue library instantiate the neopixel library using the CLUE_PIN constant, in order to tell it to never send its light values to the CPX, and that it only needs to send to the clue. If the user manually imported neopixel to use with the CLUE, the neopixel library wouldn't have CLUE_PIN for its gpio pin and it would send to both devices. 

**TLDR; the CLUE_PIN is only used when the clue library instantiates a neopixel instance. The CPX never instantiates a neopixel instance because it was built using its own neopixel library. I am hoping that we can integrate the actual neopixel library with the CPX sometime in the future, but the task seems a little out of bounds for this PR**


Sorry for the long read 😥😥

Reply:
No worries, thanks for going through this! Just so we can keep track of this, I'd recommend adding a comment or creating a work item/issue on GitHub about that. This should be good for this PR though.